rts/Rendering/SmoothHeightMeshDrawer.cpp

```diff
@@ -1,18 +1,86 @@
 /* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
 
 #include "SmoothHeightMeshDrawer.h"
-#include "Sim/Misc/SmoothHeightMesh.h"
+
+#include "Game/UI/MiniMap.h"
+#include "Map/ReadMap.h"
+#include "Rendering/GL/myGL.h"
 #include "Rendering/GL/VertexArray.h"
+#include "Sim/Misc/GlobalSynced.h"
+#include "Sim/Misc/SmoothHeightMesh.h"
+#include "System/EventHandler.h"
 #include "System/float3.h"
 
+using namespace SmoothHeightMeshNamespace;
+
+static SmoothHeightMeshDrawer* smoothMeshDrawer = nullptr;
 
 SmoothHeightMeshDrawer* SmoothHeightMeshDrawer::GetInstance() {
-	static SmoothHeightMeshDrawer drawer;
-	return &drawer;
+	if (smoothMeshDrawer == nullptr) {
+		smoothMeshDrawer = new SmoothHeightMeshDrawer();
+	}
+	return smoothMeshDrawer;
+}
+
+void SmoothHeightMeshDrawer::FreeInstance() {
+	if (smoothMeshDrawer != nullptr) {
+		delete smoothMeshDrawer;
+		smoothMeshDrawer = nullptr;
+	}
+}
```

Comment:
Because I added a destructor for unregistering for system events. I was in two minds about the approach, but that seemed to be how it was done where I picked the code up from. I was expecting to run into a problem when loading up a second or third match and would need to ensure the object got destroyed and created again... Though that didn't seem to happen.

Reply:
Ah makes sense, thanks!